Pyrado/scripts/training/qq-su_sac.py

```diff
@@ -0,0 +1,107 @@
+# 3. Neither the name of Fabio Muratore, Honda Research Institute Europe GmbH,
+#    or Technical University of Darmstadt, nor the names of its contributors may
+#    be used to endorse or promote products derived from this software without
+#    specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL FABIO MURATORE, HONDA RESEARCH INSTITUTE EUROPE GMBH,
+# OR TECHNICAL UNIVERSITY OF DARMSTADT BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""
+Train an agent to solve the Quanser Qube swing-up task using Soft Actor Critic.
+"""
+import torch as to
+from torch.optim import lr_scheduler
+
+import pyrado
+from pyrado.algorithms.step_based.sac import SAC
+from pyrado.environment_wrappers.action_normalization import ActNormWrapper
+from pyrado.environments.pysim.quanser_qube import QQubeSwingUpSim
+from pyrado.logger.experiment import save_dicts_to_yaml, setup_experiment
+from pyrado.policies.feed_back.fnn import FNNPolicy
+from pyrado.policies.feed_back.two_headed_fnn import TwoHeadedFNNPolicy
+from pyrado.spaces import ValueFunctionSpace
+from pyrado.spaces.box import BoxSpace
+from pyrado.utils.argparser import get_argparser
+from pyrado.utils.data_types import EnvSpec
+
+
+if __name__ == "__main__":
+    # Parse command line arguments
+    args = get_argparser().parse_args()
+    seed_str = f"seed-{args.seed}" if args.seed is not None else None
+
+    # Experiment (set seed before creating the modules)
+    ex_dir = setup_experiment(QQubeSwingUpSim.name, f"{SAC.name}", seed_str)
+
+    # Set seed if desired
+    pyrado.set_seed(args.seed, verbose=True)
+
+    # Environment
+    env_hparams = dict(dt=1 / 100.0, max_steps=600)
+    env = QQubeSwingUpSim(**env_hparams)
+    env = ActNormWrapper(env)
+
+    # Policy
+    policy_hparam = dict(shared_hidden_sizes=[64, 64], shared_hidden_nonlin=to.relu)
+    policy = TwoHeadedFNNPolicy(spec=env.spec, **policy_hparam)
+
+    # Q-Functions
+    qfnc_param = dict(hidden_sizes=[64, 64], hidden_nonlin=to.relu)
+    combined_space = BoxSpace.cat([env.obs_space, env.act_space])
+    q1 = FNNPolicy(spec=EnvSpec(combined_space, ValueFunctionSpace), **qfnc_param)
+    q2 = FNNPolicy(spec=EnvSpec(combined_space, ValueFunctionSpace), **qfnc_param)
+
+    # Algorithm
+    algo_hparam = dict(
+        max_iter=50,
+        memory_size=1000000,
```

Comment:
maybe we want to make this relative to env.max_steps

Reply:
Sure, I did not yet evaluate how memory size affects performance if the `env.max_steps` changes. Will check and report back

Reply:
Seems that setting this to about ten million is both often used for this kind of problem and works quite well here. In this specific task it might be also sufficient to "only" use a buffer of about three million but I see no point in restricting this in this way.